src/libs/ajax/leonardo/Disks.ts

```diff
@@ -13,12 +13,13 @@ export const Disks = (signal) => {
   });
 
   const v1Func = () => ({
-    list: async (labels = {}): Promise<ListDiskItem[]> => {
+    list: async (labels = {}): Promise<DecoratedPersistentDisk[]> => {
```

Comment:
This is where Nick and Miguel would be proposing a change.

I don't know if we want to implement this in a way that's fully compliant with the recommendation of the cross-team proposal, but for the future, here's my understanding of what that would entail:

In [the long discussion on the other sanitization PR](https://github.com/DataBiosphere/terra-ui/pull/4110), to fit with the structure proposed in the [code sharing proposal](https://docs.google.com/document/d/1vefkF3tbIPgn_2Kbx6lscFK6nYoZoaowlkXXkszxijA/edit#heading=h.cmco44cpjoai), this method would remain unchanged, returning `ListDiskItem[]`. This Ajax code is the Data Client layer.

Above that we would add two new classes: a Data Provider and a State Provider. Above that would be the Components, which would interact only with State Providers. (Right now our Components are directly calling Ajax.)

This is how I understand these new layers to work. The best example I can find is `AnalysisFile`, which is a type used in the Component `ExportAnalysisModal.ts`. It's unfortunately not a perfectly straightforward example, and I'm sure if I were better at React I'd be able to explain it more simply, but for now this is the best I can do.

So in the case of the AnalysisFile[] data type:

### Data Client fetches from the server with Ajax calls
* Both `GoogleStorage.listAnalyses()` and `AzureStorage.listNotebooks()` return AnalysisFile[] objects. This is the raw data from the server. The Data Client layer makes raw HTTP calls and handles headers, JSON, and responses, returning unsanitized objects.

### Provider layer wraps the Data Client
* `AnalysisProvider` is responsible for calling the Data Client (the Ajax methods). It has a `list` method which takes params and returns `AnalysisFile[]`. This is where the data is sanitized - the Ajax list actions (listAnalyses and listNotebooks) return GCP and Azure files, and the Provider unifies them as AnalysisFiles. The State Provider only needs to know about AnalysisFiles, and doesn't need any logic to distinguish GCP from Azure.

### State Provider layer wraps the Provider
* `useAnalysisExportState` sets up some of the wiring needed around the Ajax fetch, like `signal`, metrics, and tracking whether requests are in progress (using the LoadedState wrapper). Most importantly it creates stateful data holders (`existingAnalysisFiles`), and updates them whenever it asks its Provider (`AnalysisProvider`) for data.
* `useState` is the [React hook](https://react.dev/learn/state-a-components-memory) that enables this - providing a getter and setter for values that we want to preserve across components.

### Component layer wraps the State Provider
Instead of calling `Disks.disksV1().list()` directly, Component callers would ask a State Provider to provide stateful data. 
* `ExportAnalysisModal` tells `useAnalysisExportState`, its State Provider, for the `existingAnalysisFiles` state. This starts off empty. When the modal is ready to get files for a workspace (as part of the copy workflow which isn't important for this example), it calls a method on the State Provider which causes `existingAnalysisFiles` to update. The modal (Component) is watching that state, and rerenders on change.

Maybe there's some kind of intermediate structure we could set up with a TODO to expand into the full architecture. We didn't scope out a refactor on this scale.

Reply:
Lets chat about this in walkthrough?